acados/ocp_qp/ocp_qp_osqp.c

```diff
@@ -658,35 +659,51 @@ void ocp_qp_osqp_opts_set(void *config_, void *opts_, const char *field, void *v
 {
     ocp_qp_osqp_opts *opts = opts_;
 
+    // NOTE/TODO(oj): options are copied into OSQP at first call.
+    // Updating options through this function does not work, only before the first call!
     if (!strcmp(field, "iter_max"))
     {
         int *tmp_ptr = value;
         opts->osqp_opts->max_iter = *tmp_ptr;
     }
     else if (!strcmp(field, "tol_stat"))
     {
-        // TODO set solver exit tolerance
+        double *tol = value;
+        // opts->osqp_opts->eps_abs = *tol;
+        opts->osqp_opts->eps_rel = *tol;
+        opts->osqp_opts->eps_dual_inf = *tol;
```

Comment:
If this is the tolerance before the polishing step, it could take a _lot_ of iterations to get there.
The most efficient way of using OSQP is to leave a rather low tolerance (as I guess the default is), and hope for the polishing step to correctly guess the active set and consequently return an high-accuracy solution based on the solution of an additional equality-constrained QP.

Reply:
As you maybe saw in the commented lines below, I already played around with that.
On the example I tested it with, I reduced the total number of QP iterations from 2500 to 1220.
Full output:
[New options](https://pastebin.com/LnH3eyZZ)
[Old](https://pastebin.com/RzB53bZT)